pyanaconda/modules/payload/live/live_os.py

```diff
@@ -99,44 +120,28 @@ def space_required(self):
         """
         return get_dir_size("/") * 1024
 
-    def detect_live_os_base_image(self):
-        """Detect live os image in the system."""
-        log.debug("Trying to detect live os base image automatically")
-        for block_device in ["/dev/mapper/live-base", "/dev/mapper/live-osimg-min"]:
-            try:
-                if stat.S_ISBLK(os.stat(block_device)[stat.ST_MODE]):
-                    log.debug("Detected live base image %s", block_device)
-                    return block_device
-            except FileNotFoundError:
-                pass
-
-        # Is it a squashfs+overlayfs base image?
-        if os.path.exists("/run/rootfsbase"):
-            try:
-                block_device = execWithCapture("findmnt", ["-n", "-o", "SOURCE", "/run/rootfsbase"]).strip()
-                if block_device:
-                    log.debug("Detected live base image %s", block_device)
-                    return block_device
-            except (OSError, FileNotFoundError):
-                pass
-
-        log.debug("No live base image detected")
-        return ""
-
-    def setup_installation_source_with_task(self):
-        """Setup installation source device."""
-        return SetupInstallationSourceTask(self.image_path, INSTALL_TREE)
-
-    def teardown_installation_source_with_task(self):
+    def setup_installation_source_with_tasks(self):
+        """Setup installation source."""
+        self._check_source_availability("Set up source failed - source is not set!")
+
+        return self._get_source.set_up_with_tasks()
+
+    def teardown_installation_source_with_tasks(self):
```

Comment:
Just a note, if someone changes sources while they are set up, they won't be teared down, because we don't remember them.

Reply:
For that I'm thinking that the source data can't be changed when source is setup. 

Second option is to make the source "immutable" from the first time when the data was set. In other words - can be set only once (or to a first set up task?).

Reply:
It might be enough to raise an exception if someone tries to change the list of sources, while they are up, to avoid problems. We can provide a less strict solution later if necessary.